File: result/metagpt/claude/humaneval/21.c
#include <stdio.h>
#include <stdlib.h>

float* rescale_to_unit(float* numbers, int size) {
    float min = 100000, max = -100000;
    
    // Find the minimum and maximum values in the array
    for (int i = 0; i < size; i++) {
        if (numbers[i] < min) min = numbers[i];
        if (numbers[i] > max) max = numbers[i];
    }
    
    // Rescale the numbers to the range [0, 1]
    for (int i = 0; i < size; i++) {
        numbers[i] = (numbers[i] - min) / (max - min);
    }
    
    return numbers;
}

File: result/metagpt/gpt4o/humaneval/21.rs
fn rescale_to_unit(numbers: &mut [f32]) {
    let mut min = f32::MAX;
    let mut max = f32::MIN;

    // Find the minimum and maximum values in the array
    for &number in numbers.iter() {
        if number < min {
            min = number;
        }
        if number > max {
            max = number;
        }
    }

    // Rescale the numbers to the range [0, 1]
    for number in numbers.iter_mut() {
        *number = (*number - min) / (max - min);
    }
}

fn main() {
    let mut numbers = vec![3.0, 6.0, 9.0, 12.0];
    rescale_to_unit(&mut numbers);
    println!("Rescaled numbers: {:?}", numbers);
}